tests/framework/BaseYiiTest.php

```diff
@@ -60,4 +62,26 @@ public function testPowered()
     {
         $this->assertTrue(is_string(Yii::powered()));
     }
+
+    public function testCreateObjectCallable()
+    {
+        Yii::$container = new Container();
+
+        // Test passing in of normal params combined with DI params.
+        $this->assertTrue(Yii::createObject(function(Singer $singer, $a) {
+            return $a === 'a';
+        }, ['a']));
+
+
+        $singer = new Singer();
+        $singer->firstName = 'Bob';
+        $this->assertTrue(Yii::createObject(function(Singer $singer, $a) {
+            return $singer->firstName === 'Bob';
+        }, [$singer, 'a']));
+
+
+        $this->assertTrue(Yii::createObject(function(Singer $singer, $a = 3) {
+            return true;
+        }));
```

Comment:
We need to test here: if `$singer` argument is populated correctly at the `Closure` invoke.
`return true` - is too generic and thus risk prone.
Even, If wrong function argument produce uncathable PHP fatal error, closure is better to return something more specific. For example:

```php
return 'argument created by DI';
```